src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js

```diff
@@ -688,7 +688,10 @@ ApiGlobe.prototype.pan = function pan(pVector) {
  * @return     {number}  The zoom level.
  */
 ApiGlobe.prototype.getZoomLevel = function getZoomLevel() {
-    return this.scene.getMap().getZoomLevel();
+    const theoricalZoom = this.scene.getMap().computeTheoricalZoom(this.scene.currentCamera(), this.getRange());
+    // const idNode = this.scene.gfxEngine.screenCoordsToNodeId();
+    // const realZoom = this.scene.getMap().getZoomLevelFromIdnode(idNode);
+    return theoricalZoom;
```

Comment:
Tell me if I'm wrong : the zoom level returned is calculated by retrieving the level of the last requested tiles in the WMTS pyramid. 
How the zoom level can be calculated in the case there is no WMTS layer added to the globe ?

- It is maybe safer to calculate the zoomLevel using the scale / resolution of the view ? (see https://geoservices.ign.fr/documentation/geoservices/wmts.html#r%C3%A9solutions---niveaux-de-zoom)

For information purposes,  OpenLayers offers a way to change the default maxResolution and zoomFactor of the view, in case layers added are in other projection that Spherical Mercator (EPSG:3857) : see https://openlayers.org/en/latest/doc/tutorials/concepts.html